### PR TITLE
Ported internal document about UCX persistence schema

### PR DESCRIPTION
**Pull request by FastLee**
Generated the table persistence document in docs

### CODE REVIEW

File: docs/table_persistence.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+| Column    | Datatype | Description | Comments |
+|-----------|----------|-------------|----------|
+| catalog   | string   | Original catalog of the table. _hive_metastore_ by default |
+| database  | string   | Original schema of the table |
+| name |string|Name of the table|
+|object_type|string|MANAGED, EXTERNAL, or VIEW|
+|table_format|string|Table provider. Like delta or json or parquet.|
+|location|string|Location of the data for table|
+|view_text|nullable string|If the table is the view, then this column holds the definition of the view|
+|workspace_id|string|A workspace ID of the HMS (Optional)|
```
Comment (nfx) [inline docs/table_persistence.md:36]:
```suggestion
```
remove this :) it's going to be added on the account level - see https://github.com/databrickslabs/ucx/pull/299/files

File: docs/table_persistence.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+#### _$inventory_.jobs
+Holds a list of all jobs with a notation of potential issues.
+
+| Column    | Datatype | Description | Comments |
+|-----------|----------|-------------|----------|
+|job_id|string|Job ID|
+|job_name|string|Job Name|
+|job_creator|string|UserID of the Job Creator|
+|compatible|int|1 or 0, used for percentage reporting|
+|notes|string|List of issues identified by the assessment in JSON format|
```
Comment (nfx) [inline docs/table_persistence.md:99]:
```suggestion
|failures|string|List of issues identified by the assessment in JSON format|
```

File: docs/table_persistence.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+|azure_sp|string|List of Azure SP utilized by this job|
```
Comment (nfx) [inline docs/table_persistence.md:100]:
```suggestion
```

we have a separate task that gets this. no need to keep it here. let's capture the inventory schema current state and then make proposals as additional PRs to this?

File: docs/table_persistence.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+#### _$inventory_.clusters
+Holds a list of all clusters with a notation of potential issues.
+
+| Column    | Datatype | Description | Comments |
+|-----------|----------|-------------|----------|
+|cluster_id|string|Cluster Id|
+|cluster_name|string|Cluster Name|
+|cluster_creator|string|UserID of the Cluster Creator|
+|compatible|int|1 or 0, used for percentage reporting|
+|notes|string|List of issues identified by the assessment in JSON format|
```
Comment (nfx) [inline docs/table_persistence.md:112]:
```suggestion
|failures|string|List of issues identified by the assessment in JSON format|
```

File: docs/table_persistence.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+|azure_sp|string|List of Azure SP utilized by this job|
```
Comment (nfx) [inline docs/table_persistence.md:113]:
```suggestion
```

File: docs/table_persistence.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+Enumeration of all UCX persistence elements
+
+## Overview
+Table Utilization:
+
+| Table              | Generate Assessment | Migrate Local Groups | Migrate External Tables | Migrate SQL Warehouses | Upgrade Jobs | Migrate managed tables |
+|--------------------|---------------------|----------------------|-------------------------|------------------------|--------------|------------------------|
+| tables             | RW                  |                      | RO                      |                        |              | RO                     |
+| grants             |                     |                      | RW                      |                        |              | RW                     |
```
Comment (nfx) [inline docs/table_persistence.md:11]:
```suggestion
| grants             |   W                 |                      | RW                      |                        |              | RO                     |
```
write-only

File: docs/table_persistence.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+| mounts             | RW                  |                      | RO                      |                        | RO           | RO                     |
+| permissions        |                     | RW                   |                         |                        |              |                        |
+| jobs               | RW                  |                      |                         |                        | RO           |                        |
+| clusters           | RW                  |
+| external_locations | RW                  |                      | RO                      |
+| workspace          | RW                  | RO                   |                         |                        | RO           |
+
+**RW** - Read/Write the job that generates the table
```
Comment (nfx) [inline docs/table_persistence.md:19]: most of the things cannot have RW.

**Review by nfx — CHANGES_REQUESTED** [review]

File: docs/table_persistence.md (hunk 7 above)
Comment (FastLee) [reply docs/table_persistence.md:19]: I'm capturing the Job that generate these.

**Review by nfx — APPROVED** [review]

**Review by nfx — APPROVED** [review]

**Pull request merged**